File: Commons/src/main/java/ca/warp7/frc/RobotController.java
package ca.warp7.frc;

@SuppressWarnings({"unused"})
public class RobotController {

    public enum State {
        Pressed, Released, HeldDown, KeptUp
    }

    State AButton;
    State BButton;
    State XButton;
    State YButton;
    State leftBumper;
    State rightBumper;
    State leftTrigger;
    State rightTrigger;
    State leftStickButton;
    State rightStickButton;
    State startButton;
    State backButton;
    State upDPad;
    State rightDPad;
    State downDPad;
    State leftDPad;

    double leftTriggerAxis;
    double rightTriggerAxis;
    double leftXAxis;
    double leftYAxis;
    double rightXAxis;
    double rightYAxis;

    public State getAButton() {
        return AButton;
    }

    public State getBButton() {
        return BButton;
    }

    public State getXButton() {
        return XButton;
    }

    public State getYButton() {
        return YButton;
    }

    public State getLeftBumper() {
        return leftBumper;
    }

    public State getRightBumper() {
        return rightBumper;
    }

    public State getLeftTrigger() {
        return leftTrigger;
    }

    public State getRightTrigger() {
        return rightTrigger;
    }

    public State getLeftStickButton() {
        return leftStickButton;
    }

    public State getRightStickButton() {
        return rightStickButton;
    }

    public State getStartButton() {
        return startButton;
    }

    public State getBackButton() {
        return backButton;
    }

    public State getUpDPad() {
        return upDPad;
    }

    public State getRightDPad() {
        return rightDPad;
    }

    public State getDownDPad() {
        return downDPad;
    }

    public State getLeftDPad() {
        return leftDPad;
    }

    public double getLeftTriggerAxis() {
        return leftTriggerAxis;
    }

    public double getRightTriggerAxis() {
        return rightTriggerAxis;
    }

    public double getLeftXAxis() {
        return leftXAxis;
    }

    public double getLeftYAxis() {
        return leftYAxis;
    }

    public double getRightXAxis() {
        return rightXAxis;
    }

    public double getRightYAxis() {
        return rightYAxis;
    }

}

File: Commons/src/main/kotlin/ca/warp7/frc/RobotController.kt
package ca.warp7.frc

@Suppress("unused")
class RobotController internal constructor() {
    internal val data = ControllerData()
    val aButton get() = data.aButton
    val bButton get() = data.bButton
    val xButton get() = data.xButton
    val yButton get() = data.yButton
    val leftBumper get() = data.leftBumper
    val rightBumper get() = data.rightBumper
    val leftStickButton get() = data.leftStickButton
    val rightStickButton get() = data.rightStickButton
    val startButton get() = data.startButton
    val leftTriggerAxis get() = data.leftTriggerAxis
    val rightTriggerAxis get() = data.rightTriggerAxis
    val leftXAxis get() = data.leftXAxis
    val leftYAxis get() = data.leftYAxis
    val rightXAxis get() = data.rightXAxis
    val rightYAxis get() = data.rightYAxis
}

